web/src/settings_invites.ts

```diff
@@ -103,9 +121,24 @@ function populate_invites(invites_data: {invites: Invite[]}): void {
                 item.expiry_date_absolute_time = timerender.absolute_time(item.expiry_date * 1000);
             }
             item.is_admin = current_user.is_admin;
-            item.disable_buttons =
-                item.invited_as === settings_config.user_role_values.owner.code &&
-                !current_user.is_owner;
+
+            // Disable edit if user cannot assign the invite's current role
+            const role = item.invited_as;
+            const role_values = settings_config.user_role_values;
+            let can_edit = false;
+            if (role === role_values.guest.code || role === role_values.member.code) {
+                can_edit = true;
+            } else if (
+                (role === role_values.moderator.code || role === role_values.admin.code) &&
+                (current_user.is_admin || current_user.is_owner)
+            ) {
+                can_edit = true;
+            } else if (role === role_values.owner.code && current_user.is_owner) {
+                can_edit = true;
+            }
+            item.disable_edit_button =
+                !can_edit || !settings_data.user_can_create_multiuse_invite();
+            item.disable_revoke_and_resend_button = item.disable_edit_button;
```

Comment:
This is not correct now.

First because only email invites can be resent, so disabling the resend button should not depend on permission to edit invite. And for revoking as well permissions for revoking email invite should not change and I don't think permission to revoke multiuse invite should depend on permission to create invites.

Also, I don't think we have discussed about changing permission to revoke invite based on `invite_as` value. I think it is correct to do so but would be good to just confirm it once.

Reply:
And then we would also need to add the restriction on revoking based on `invite_as` value in backend as well. Would be good to have all changes for revoking permission in a separate commit.